sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,132 @@
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
+#include "opentelemetry/sdk/trace/samplers/always_on.h"
+#include "opentelemetry/sdk/trace/samplers/parent_or_else.h"
+#include "opentelemetry/sdk/trace/samplers/probability.h"
+
+#include <cstdint>
+
+#include <benchmark/benchmark.h>
+
+namespace
+{
+using opentelemetry::sdk::trace::AlwaysOffSampler;
+using opentelemetry::sdk::trace::AlwaysOnSampler;
+using opentelemetry::sdk::trace::ParentOrElseSampler;
+using opentelemetry::sdk::trace::ProbabilitySampler;
+using opentelemetry::sdk::trace::Decision;
+using opentelemetry::trace::SpanContext;
+
+// NOOP sampler constructor used as a baseline to compare with other samplers
+void BM_AlwaysOffSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOffSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOffSamplerConstruction);
+
+// NOOP sampler constructor used as a baseline to compare with other samplers
+void BM_AlwaysOnSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(AlwaysOnSampler());
+	}
+}
+BENCHMARK(BM_AlwaysOnSamplerConstruction);
+
+void BM_ParentOrElseSamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ParentOrElseSampler(std::make_shared<AlwaysOnSampler>()));
+	}
+}
+BENCHMARK(BM_ParentOrElseSamplerConstruction);
+
+void BM_ProbabilitySamplerConstruction(benchmark::State &state)
+{
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(ProbabilitySampler(0.01));
+	}
+}
+BENCHMARK(BM_ProbabilitySamplerConstruction);
+
+// NOOP sampler used as a baseline to compare with other samplers
+void BM_AlwaysOffSamplerShouldSample(benchmark::State &state)
+{
+	AlwaysOffSampler sampler;
+
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	while (state.KeepRunning())
+	{
+		benchmark::DoNotOptimize(sampler.ShouldSample(nullptr, trace_id, "", span_kind, view));
+	}
```

Comment:
This code is repeated for every sampler benchmark, can you pull it out into a helper?

Reply:
Yep, I've went ahead and done that.